src/application/application_api.js

```diff
@@ -94,23 +105,181 @@ function defineReadOnlyProperty(object, key, value) {
   });
 }
 
-// Define Application interface, the getRequestedAppControl method will not be
-// implemented ATM.
+function defineNonNullProperty(object, key, value) {
+  var hvalue = value;
+  Object.defineProperty(object, key, {
+    enumerable: true,
+    set: function(val) {
+      if (val !== null && val !== undefined) {
+        hvalue = val;
+      }
+    },
+    get: function() {
+      return hvalue;
+    }
+  });
+}
+
+// ApplicationControlData interface
+tizen.ApplicationControlData = function(key, value) {
+  if (!(this instanceof tizen.ApplicationControlData)) {
+    throw new TypeError();
+  }
+
+  if (typeof key !== 'string' ||
+      !(value instanceof Array)) {
+    throw new tizen.WebAPIException(tizen.WebAPIException.TYPE_MISMATCH_ERR);
+  }
+
+  // FIXME: tests assume you can push anything to value
+  // so stringify data instead of validating it
+  for (var i = 0; i < value.length; i++) {
+    value[i] = String(value[i]);
+  }
+
+  defineNonNullProperty(this, 'key', key);
+  defineNonNullProperty(this, 'value', value);
+};
+
+function toApplicationControlDataArray(data) {
+  var array_objects = [];
+  for (var i = 0; i < data.length; i++) {
+    if (typeof data[i]['key'] != 'string' ||
+        !(data[i]['value'] instanceof Array)) {
```

Comment:
nit, squash into one line, you can check `tools/check-coding-style`, we use 100 chars